tensorflow/python/data/experimental/kernel_tests/data_service_ops_test.py

```diff
@@ -648,6 +648,86 @@ def testRegisterDifferentDatasets(self):
     id_2 = data_service_ops.register_dataset(cluster.target, ds_2)
     self.assertNotEqual(id_1.numpy(), id_2.numpy())
 
+  @combinations.generate(test_base.eager_only_combinations())
+  def testDistributeParallelToDistributed(self):
+
+    cluster_1 = self.create_cluster(num_workers=1)
+    cluster_2 = self.create_cluster(num_workers=1)
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds = self.make_distributed_dataset(
+        ds, cluster_1, processing_mode="parallel_epochs")
+    ds = ds.map(lambda x: x + 1)
+    ds = self.make_distributed_dataset(
+        ds, cluster_2, processing_mode="distributed_epoch")
+
+    error_regex = "Cannot create a split provider for dataset of type DataServiceDataset"  # pylint: disable=line-too-long
+    with self.assertRaisesRegex(errors.UnimplementedError, error_regex):
+      self.getDatasetOutput(ds)
+
+  @combinations.generate(test_base.eager_only_combinations())
+  def testDistributeDistributedToParallel(self):
+
+    cluster_1 = self.create_cluster(num_workers=1)
+    cluster_2 = self.create_cluster(num_workers=1)
+    num_sizes = 10
+    size_repeats = 5
+    numbers = [1 * i for i in range(num_sizes)] * size_repeats
+    ds = dataset_ops.Dataset.from_tensor_slices(numbers)
+    ds = self.make_distributed_dataset(
+        ds, cluster_1, processing_mode="distributed_epoch")
+    ds = ds.map(lambda x: x + 1)
+    ds = self.make_distributed_dataset(
+        ds, cluster_2, processing_mode="parallel_epochs")
+
+    self.assertDatasetProduces(
+        ds, [i + 1 for i in numbers], assert_items_equal=True)
+
+  def testParallelZippedDistributedDatasets(self):
```

Comment:
@aaudiber apologies if the name of the test is confusing. However, since I didn't find any test that tries to distribute a zipped dataset which was prepared using already distributed datasets, I thought of adding this. Will remove it if you feel that this is redundant.

Reply:
No worries. I think we need to consider the cost of maintaining and running these tests during the build, so it's not worth adding this specific test when there are already such similar tests

Reply:
I agree. Removed the redundant test.